src/ui/projectmanager/pm.go

```diff
@@ -18,15 +18,43 @@ import (
 	"github.com/vmware/harbor/src/common/models"
 )
 
+// QueryParam can be used to set query parameters when listing projects
+type QueryParam struct {
+	Name       string      // the name of project
+	Owner      string      // the username of project owner
+	Public     string      // the project is public or not, can be "ture","false" and ""
```

Comment:
I'm not comfortable with this, essentially we need a union type here.  If you don't want to implement a interface with limited "isxxx" function, maybe a pointer to a bool? so it has three condition, and can't be wrong?  Also explains what will happen is it's nil.